### PR TITLE
ansible: Add task runners in PSI OpenStack

### PR DESCRIPTION
**Pull request by martinpitt**
We recently got a "front-door-ci" project on the rhos-d and rhos-01
OpenStack clouds. The latter's nested KVM support and performance is
good enough for our tasks runners (on par with e2e machines).

Add a "launch-tasks" playbook to the new ansible/psi/ directory for 
creating a single instance. We don't have very big flavors there, so
each instance can run just one tasks bot. Create these instances in the 
"openstack_tasks" group, so that we can refer to them as an Ansible
hosts group.

Install Fedora 35 on them for the time being, as we don't currently have
CoreOS images in that cloud.

Add dynamic OpenStack inventory for PSI rhos-01.  This will pick up all 
hosts in PSI rhos-01 for the inventory.  Extend the tasks maintenance
playbooks to also run in the "openstack_tasks" group. See 

    ansible-inventory -i inventory/openstack.yml -v --yaml --list

for available hosts and properties.

-----

I created three task runners, and will observe them through the next few days. I've had two manually configured test instances for three days, which performed well.

Runs on the new infra: [log](https://cockpit-logs.us-east-1.linodeobjects.com/pull-3616-20220715-065914-b2262694-rhel-9-1-cockpit-project-cockpit/log.html), [log](https://cockpit-logs.us-east-1.linodeobjects.com/pull-3616-20220715-041236-b61021bb-rhel-9-1-candlepin-subscription-manager/log.html), [log](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17441-20220718-075938-67b50661-fedora-36-firefox/log.html), [log](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17571-20220718-095624-44b4461f-rhel-8-7/log.html#246-2), [log](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1042-20220719-043850-23edf756-fedora-coreos/log.html)

### CODE REVIEW

**Comment by martinpitt** [comment]
@marusak : It would be great if you could try this -- i.e. follow the README, point out what's unclear, check that you have access. Please create a 4th instance, SSH to it, and check that `sudo journalctl -f -u cockpit-tasks@1` starts up properly. Thanks!

**Comment by martinpitt** [comment]
Ugh @ "sd-bus call: Connection reset by peer" in the [tasks test](https://github.com/cockpit-project/cockpituous/runs/7337179255?check_suite_focus=true). This is obviously unrelated, but independently needs looking into. /me yells at current pilot (I think I've seen something similar while wrestling with cockpit's unit tests refresh)

**Comment by martinpitt** [comment]
I brought up some more instances, as I want to shift some load from e2e to PSI. Today's instances failed to start the container:

    podman[13296]: /usr/bin/crun: symbol lookup error: /usr/bin/crun: undefined symbol: criu_join_ns_add

Updating installed packages first helps. (Argh, I *really* want CoreOS back!).

I updated the playbook to install package updates, and rolled out some new instances with that.

**Comment by martinpitt** [comment]
I shifted some more loads to PSI: I stopped our bots on ci-srv-* and created rhos-01-{4..12}. @marusak, when you try this, please create -13 and -14 instead of -4.

File: ansible/psi/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Resources for setting up CI in PSI OpenStack
+============================================
+
+We have a "front-door-ci" project on the PSI OpenStack clouds rhos-d and rhos-01. The latter's nested KVM support and performance is good enough for our tasks runners.
+
+Access
+------
+Sign into the OpenShift web console on https://api.rhos-01.prod.psi.rdu2.redhat.com with group "redhat.com", your RedHat user name, and your Kerberos password. You need to be in the [front-door-ci group](https://rover.redhat.com/groups/group/front-door-ci) for this.
```
Comment (marusak) [inline ansible/psi/README.md:8]: what does it mean "group readhat.com"? does that refer to "Domain" field?

Comment (marusak) [reply ansible/psi/README.md:8]: Apparently, can we update that?

File: ansible/psi/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+The only change that was, and currently needs to be done manually is to add SSH and ICMP to the "default" security group. This step may be automated in the future.
+
+Automation setup
+----------------
+
+You need to install the OpenStack SDK, either with `sudo dnf install python3-openstacksdk` (on Fedora), or `pip install openstacksdk`.
+
+After that, you need to set up `~./config/openstack/clouds.yml`. If you don't already have one, copy or symlink [clouds.yml](./clouds.yml), otherwise merge it with your existing one. You need to add your credentials to `~/.config/openstack/secure.yaml`, see the comment in [clouds.yml](./clouds.yml) for how to do that.
```
Comment (marusak) [inline ansible/psi/README.md:17]: it should be `~/.config/openstack/clouds.yml` (dot is before `/`)

File: ansible/psi/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Access
+------
+Sign into the OpenShift web console on https://api.rhos-01.prod.psi.rdu2.redhat.com with group "redhat.com", your RedHat user name, and your Kerberos password. You need to be in the [front-door-ci group](https://rover.redhat.com/groups/group/front-door-ci) for this.
+
+The only change that was, and currently needs to be done manually is to add SSH and ICMP to the "default" security group. This step may be automated in the future.
+
+Automation setup
+----------------
+
+You need to install the OpenStack SDK, either with `sudo dnf install python3-openstacksdk` (on Fedora), or `pip install openstacksdk`.
```
Comment (marusak) [inline ansible/psi/README.md:15]: I also need `python3-openstackclient`

File: ansible/psi/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+After that, you need to set up `~./config/openstack/clouds.yml`. If you don't already have one, copy or symlink [clouds.yml](./clouds.yml), otherwise merge it with your existing one. You need to add your credentials to `~/.config/openstack/secure.yaml`, see the comment in [clouds.yml](./clouds.yml) for how to do that.
+
+Check that this works with
+
+    openstack --os-cloud rhos-01 server list -n
+    ansible-inventory -i inventory/openstack.yml -v --yaml --list
```
Comment (marusak) [inline ansible/psi/README.md:22]:
```
Using /etc/ansible/ansible.cfg as config file
[WARNING]: Couldn't list Openstack hosts. See logs for details
Invalid input for field 'identity/password/user/password': None is not of type 'string'

Failed validating 'type' in schema['properties']['identity']['properties']['password']['properties']['user']['properties']['password']:
    {'type': 'string'}

On instance['identity']['password']['user']['password']:
    None (HTTP 400) (Request-ID: req-90b8c377-8894-4d07-962c-9c47aa0c46f8)
all:
  children:
    ungrouped: {}
```

Do I need to log in? First command worked with `--os-username mmarusak` and then passing password

File: ansible/psi/README.md (hunk 2 above)
Comment (marusak) [reply ansible/psi/README.md:17]: While editing this, can you please rewrite `You also need to add your credentials to...` (adding `also`) as first I misunderstood that you only need to do it if you already have `clouds.yml`. Or put it on separate line?

File: ansible/psi/README.md (hunk 3 above)
Comment (martinpitt) [reply ansible/psi/README.md:15]: I don't have that installed. It's not necessary for running the playbooks, you just need it if you want to run the `openstack` CLI command directly. I'll instead drop the `openstack` mention in README, it's redundant anyway.

File: ansible/psi/README.md (hunk 1 above)
Comment (martinpitt) [reply ansible/psi/README.md:8]: Right, done.

File: ansible/psi/README.md (hunk 2 above)
Comment (martinpitt) [reply ansible/psi/README.md:17]: You mean the two "clouds.yml" links? The link targets are supposed to lead to the template in git. See https://github.com/cockpit-project/cockpituous/blob/psi-openstack/ansible/psi/README.md , they seem to be working as intended. Or is there something wrong with my `~./config/openstack/clouds.yml`? (then I don't see it, sorry)

**Comment by marusak** [comment]
I have created worker -13 and it seems to work just fine :tada: 

I see this warning, but not sure where it comes from:
```
Not replacing invalid character(s) "{'-'}" in group name (rhos-01)
[WARNING]: Invalid characters were found in group names but not replaced, use -vvvv to see details
```

**Review by marusak — APPROVED** [review]
All works, thanks!

**Pull request merged**